pkg/services/preferences/preferences.go

```diff
@@ -0,0 +1,76 @@
+package preferences
+
+import (
+	"context"
+	"strings"
+
+	"github.com/grafana/grafana/pkg/models"
+	pstore "github.com/grafana/grafana/pkg/services/preferences/store"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+type Service interface {
+	GetPreferencesWithDefaults(ctx context.Context, query *models.GetPreferencesWithDefaultsQuery) (*models.Preferences, error)
```

Comment:
I think `Preferences` can removed so it is just `GetWithDefaults`, externally ends up as (with previous suggestion, `prefs.GetWithDefaults`.